frontend/catalyst/passes.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains public API functions that provide control for the
+user to input what MLIR compiler passes to run.
+
+Currently, each pass has its own user-facing decorator. In the future,
+a unified user interface for all the passes will be available.
+
+.. note::
+
+    Unlike PennyLane :doc:`circuit transformations <introduction/compiling_circuits>`,
+    the QNode itself will not changed or transformed.
+
+    In other words, circuit inspection tools such as
+    :func:`~.draw` will still
+    display the neighbouring self-inverse gates. However, Catalyst never
+    executes the PennyLane code directly; instead, Catalyst captures the
+    workflow from Python and lowers it into MLIR, performing compiler
+    optimizations at the MLIR level.
+    To inspect the compiled MLIR from Catalyst, use
+    :func:`~.print_compilation_stage`,
+    where ``stage="QuantumCompilationPass"``, and with ``keep_intermediate=True``
+    in the ``qjit`` decorator.
+
+"""
+
+import pennylane as qml
+
+from catalyst.jax_primitives import apply_registered_pass_p, transform_named_sequence_p
+
+
+def _inject_transform_named_sequence():
+    """
+    Inject a transform_named_sequence jax primitive.
+
+    This must be called when preprocessing the traced function in QJIT.capture(),
+    since to invoke -transform-interpreter, a transform_named_sequence primitive
+    must be in the jaxpr.
+    """
+
+    transform_named_sequence_p.bind()
```

Comment:
Minor, but the restructured modules have started placing the API at the so it's easiest to find. Putting helper functions first is not actually needed unless programming in C 😅

Reply:
done